include/nbl/video/alloc/SubAllocatedDescriptorSet.h

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_VIDEO_SUB_ALLOCATED_DESCRIPTOR_SET_H_
+#define _NBL_VIDEO_SUB_ALLOCATED_DESCRIPTOR_SET_H
+
+#include "nbl/video/alloc/IBufferAllocator.h"
+
+#include <type_traits>
+
+namespace nbl::video
+{
+
+class SubAllocatedDescriptorSet : public core::IReferenceCounted 
+{
+public:
+	// address allocator gives offsets
+	// reserved allocator allocates memory to keep the address allocator state inside
+	using AddressAllocator = core::GeneralpurposeAddressAllocator<uint32_t>;
+	using ReservedAllocator = core::allocator<uint8_t>;
+	using size_type = typename AddressAllocator::size_type;
+	using value_type = typename AddressAllocator::size_type;
+	static constexpr value_type invalid_value = AddressAllocator::invalid_address;
+
+protected:
+	struct SubAllocDescriptorSetRange {
+		std::shared_ptr<AddressAllocator> addressAllocator;
+		std::shared_ptr<ReservedAllocator> reservedAllocator;
```

Comment:
unique_ptr can work just as well

Reply:
you want to slap a `MultiTimelineEventHandlerST<DeferredFreeFunctor>` per binding here as well